raft/raft.go

```diff
@@ -1602,6 +1594,25 @@ func (r *raft) abortLeaderTransfer() {
 	r.leadTransferee = None
 }
 
+func (r *raft) CommittedEntryInCurrentTerm() bool {
+	return r.raftLog.zeroTermOnErrCompacted(r.raftLog.term(r.raftLog.committed)) == r.Term
+}
+
+func (r *raft) responseReadIndex(req pb.Message, committed uint64) (resp pb.Message, internal bool) {
```

Comment:
rename to ResponseToReadIndexReq?

1. why do you need to pass in the committed index? you can get it from the method receiver itself?

2. what does internal mean in the return?

Reply:
> why need to pass in the committed index
For leader, the index should be its committed index when it receiving the `ReadIndex` request, not its committed index after it gets enough heartbeat responses. So we must pass it into the function.